### PR TITLE
Unify the references for the test projects

### PR DESCRIPTION
**Pull request by pgrawehr**
This uses a common place to include the default references required for unit test projects (xunit, Microsoft.NET.Test.Sdk, Moq, etc.)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1558)

### CODE REVIEW

**Review by Ellerbach — APPROVED** [review]
Good initiative and clean up. Clearly easier like that

File: src/System.Device.Gpio.Tests/System.Device.Gpio.Tests.csproj [hunk 1]
```diff
@@ -4,14 +4,12 @@
...
-  <ItemGroup>
-    <PackageReference Include="Moq" Version="4.16.1" />
-    <PackageReference Include="xunit.runner.utility" Version="2.4.1" />
-  </ItemGroup>
 
   <ItemGroup>
     <PackageReference Include="UnitsNet" Version="$(UnitsNetPackageVersion)" />
     <ProjectReference Include="$(MainLibraryPath)System.Device.Gpio.csproj" />
     <ProjectReference Include="..\Iot.Device.Bindings\Iot.Device.Bindings.csproj" />
+     <PackageReference Include="Moq" Version="$(MoqPackageVersion)" />
```
Comment (joperezr) [inline src/System.Device.Gpio.Tests/System.Device.Gpio.Tests.csproj:12]: NIT: extra spacing here

**Comment by pgrawehr** [comment]
@joperezr I think server ddvsorpil022 should be rebooted. IIRC it was mostly involved in failing builds.

**Comment by pgrawehr** [comment]
@joperezr : I'll provide a separate PR to update the binding template, but I'm fighting to exclude that project from the build. So I'll leave that for later.

File: src/devices/Directory.Build.props [hunk 2]
```diff
@@ -16,8 +17,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="UnitsNet" Version="$(UnitsNetPackageVersion)" />
-    <PackageReference Include="SixLabors.ImageSharp" Version="$(SixLaborsImageSharpPackageVersion)" />
     <ProjectReference Include="$(SystemDeviceModelPath)" Condition="'$(MSBuildProjectName)' != '$(SystemDeviceModelProjectName)'" />
+    <ProjectReference Include="$(MainLibraryPath)/System.Device.Gpio.csproj" />
```
Comment (joperezr) [inline src/devices/Directory.Build.props:21]: Why do we now include this project Reference here? Shouldn't this be includded by each device binding?

**Review by joperezr — APPROVED** [review]

**Pull request merged**